lib/stdlib/src/gen_server.erl

```diff
@@ -223,6 +234,38 @@ call(Name, Request, Timeout) ->
 	    exit({Reason, {?MODULE, call, [Name, Request, Timeout]}})
     end.
 
+%% -----------------------------------------------------------------
+%% Make an async_call to a generic server.
+%% and return a Key (promise) which can/should be used with yield/[1|2]
+
+-spec async_call(Name::server_ref(), Request::term()) -> promise().
+async_call(Name, Request) ->
+    try gen:async_call(Name, '$gen_call', Request)
+    catch _:Reason ->
+            erlang:error({Reason, {?MODULE, ?FUNCTION_NAME, [Name, Request]}})
+    end.
+
+-spec yield(Promise::promise()) -> {reply, Reply::term()}.
+yield(Promise) ->
+    try gen:yield(Promise, infinity)
```

Comment:
Should this reflect the other timeouts in the GenServer module? I.e. 5000?

Reply:
Good question.
I don't like the 5s timeout but now it's there.
Opinions?

Reply:
Yeah, I hear you. :/

I am worried someone may start with `call`, then move to `async_call`+`yield` as some optimization but by doing so they have changed the meaning of the code in an unexpected way.

Reply:
I think I agree with José. All other timeouts are 5s, then this should be as well.

Reply:
though, really if what would be the same thing would be if async_call would setup a timer for 5 seconds when it is called and then yield/yield_message would timeout when they receive the timer. Don't think we want to have that though.

Reply:
I think that the "async_call" leaves all timeout logic up the caller, and there should be no timeout option.

Reply:
Timer would not be set in `async_call` but in `yield/2` which contains the `receive`

Reply:
The timer argument lets the user do something like getting the current time `StartTime`, do the async call, do something else and then at the end call `{ok, Result} = yield(Promise, max(0, Timeout + CurrentTime - StartTime))`. Only it would `{badarg,timeout}` instead of `exit(timeout)` but that's probably not a big deal?